### DISCUSSION TITLE
Creating a Class Inherited

### DISCUSSION THREAD

**Discussion by toto45638** [microsoft/microsoft-ui-xaml#10055] [Q&A]
I am trying to create a custom class for the ItemsPanelTemplate to customize the layout inside a ListBox.
However, when I run it, an error occurs.
It seems the error occurs in the MeasureOverride method of the winrt::Microsoft::UI::Xaml::IFrameworkElementOverrides class, not in the custom class itself. Could there be something wrong with the way I am creating the custom class?

Error

> "Exception thrown at 0x00007FFFFF9BF645 (Microsoft.ui.xaml.dll) in SampleApp.exe: 0xC0000005: Access violation reading location 0x000001AAB6153011."

idl
``` 
namespace SampleApp
{
    [default_interface]
    runtimeclass SampleGridPanel : Microsoft.UI.Xaml.Controls.Panel
    {
        SampleGridPanel();
    }
}
``` 

``` 
#pragma once
#include "SampleGridPanel.g.h"

namespace winrt::SampleApp::implementation
{
    struct SampleGridPanel : SampleGridPanelT<SampleGridPanel, Microsoft.UI.Xaml.Controls.Panel>
    {
        SampleGridPanel();

        winrt::Windows::Foundation::Size MeasureOverride(winrt::Windows::Foundation::Size availableSize);
        winrt::Windows::Foundation::Size ArrangeOverride(winrt::Windows::Foundation::Size finalSize);
    };
}

namespace winrt::SampleApp::factory_implementation
{
    struct SampleGridPanel : SampleGridPanelT<SampleGridPanel>
    {
    };
}
```

**Comment by DarranRowe** [comment]
What is the definition of that runtime class in IDL?
Also, is this defined in the executable? Have you tried creating a separate component and placing it in that? I think the Xaml compiler had at times when it was defined as part of the executable.

**Reply by toto45638** [reply]
The IDL file was missing, so I added it to the question.
The #include 'SampleGridPanel.h' is set in MainWindow.xaml.h and App.xaml.h.
The build succeeded, but an error occurred during execution.
I created a new project and tried again, but the error still occurs during execution. If I do not write <local:SampleGridPanel />, the error does not occur and the project runs normally, so it seems that there is still a problem with SampleGridPanel.

**Reply by toto45638** [reply]
I found the cause of the error. The issue was in the header file's struct definition. The Microsoft.UI.Xaml.Controls.Panel part in struct SampleGridPanel : SampleGridPanelT<SampleGridPanel, Microsoft.UI.Xaml.Controls.Panel> was unnecessary.
